test/functional/collection_tests.js

```diff
@@ -12,63 +11,64 @@ describe('Collection', function() {
     return setupDatabase(this.configuration);
   });
 
-  /**
-   * @ignore
-   */
-  it('should correctly execute basic collection methods', {
-    metadata: {
-      requires: { topology: ['single', 'replicaset', 'sharded', 'ssl', 'heap', 'wiredtiger'] }
-    },
-
-    // The actual test we wish to run
-    test: function(done) {
-      var self = this;
-      var client = self.configuration.newClient(self.configuration.writeConcernMax(), {
+  describe('', function() {
+    let client;
+    let db;
+    beforeEach(function() {
+      client = this.configuration.newClient(this.configuration.writeConcernMax(), {
         poolSize: 1
       });
-      client.connect(function(err, client) {
-        var db = client.db(self.configuration.db);
+      return client.connect().then(client => {
+        db = client.db(this.configuration.db);
+      });
+    });
+    afterEach(function() {
+      return client.close();
+    });
 
-        db.createCollection('test_collection_methods', function(err, collection) {
+    /**
+     * @ignore
+     */
+    it('should correctly execute basic collection methods', {
+      metadata: {
+        requires: { topology: ['single', 'replicaset', 'sharded', 'ssl', 'heap', 'wiredtiger'] }
+      },
+
+      // The actual test we wish to run
+      test: function(done) {
+        db.createCollection('test_collection_methods', (err, collection) => {
           // Verify that all the result are correct coming back (should contain the value ok)
-          test.equal('test_collection_methods', collection.collectionName);
+          expect('test_collection_methods').to.equal(collection.collectionName);
           // Let's check that the collection was created correctly
-          db.listCollections().toArray(function(err, documents) {
-            var found = false;
-            documents.forEach(function(document) {
-              if (document.name === 'integration_tests_.test_collection_methods') found = true;
-            });
-            test.ok(true, found);
-
+          db.listCollections().toArray(err => {
+            expect(err).to.not.exist;
```

Comment:
Kept the test in but it currently fails because the only document names in the collection are `test_collection_methods4` and `test_collection_methods3`

Reply:
Oh right, I should've read that more carefully! I forgot that was the `test.ok(true, found)` issue. We can change the check to `if (document.name === 'test_collection_methods')`. If we put that check on line 36, it should pass.

Reply:
I moved it to line 36 but the test still fails. The document names are: `test_collection_methods`, `uncapped` and `uncapped2`. Could I change the if statement to if (doc.name === 'test_collection_methods')? That seems like the proper behavior and it passes the test.

Reply:
Yes, that would be great!